examples/hotel_ads/add_performance_max_for_travel_goals_campaign.rb

```diff
@@ -0,0 +1,804 @@
+      puts "A default image URL '#{default_uri}' is used to create an image asset for the asset field type '#{asset_field_type}'."
+
+      operations +=
+        create_and_link_image_asset_operations(
+          client,
+          customer_id,
+          default_uri,
+          asset_field_type,
+          "Default image asset for the asset field type '%s'." %
+            asset_field_type
+        )
+    end
+  end
+
+  operations
+end
+
+# Creates a list of MutateOperations that create a new linked image asset.
+def create_and_link_image_asset_operations(
+  client,
+  customer_id,
+  url,
+  field_type,
+  asset_name
+)
+  operations = []
+  temp_id = next_temp_id
+
+  # Create the Image Asset.
+  operations << client.operation.mutate do |m|
+    m.asset_operation =
+      client.operation.create_resource.asset do |a|
+        a.resource_name = client.path.asset(customer_id, temp_id)
+        # Provide a unique friendly name to identify your asset.
+        # When there is an existing image asset with the same content but a different
+        # name, the new name will be dropped silently.
+        a.name = asset_name
+        a.type = :IMAGE
+        a.image_asset =
+          client.resource.image_asset do |image_asset|
+            image_asset.data = get_image_bytes(url)
+          end
+      end
+  end
+
+  # Create an AssetGroupAsset to link the Asset to the AssetGroup.
+  operations << client.operation.mutate do |m|
+    m.asset_group_asset_operation =
+      client.operation.create_resource.asset_group_asset do |aga|
+        aga.field_type = field_type
+        aga.asset_group =
+          client.path.asset_group(customer_id, ASSET_GROUP_TEMPORARY_ID)
+        aga.asset = client.path.asset(customer_id, temp_id)
+      end
+  end
+
+  operations
+end
+
+# Loads image data from a URL.
+def get_image_bytes(url)
+  URI.open(url).read
+end
+
+# Prints the details of a MutateGoogleAdsResponse.
+def print_response_details(response)
+  # Parse the mutate response to print details about the entities that
+  # were created by the request.
+  suffix = '_result'
+  response.mutate_operation_responses.each do |result|
+    result
+      .to_h
+      .select { |_k, v| v }
+      .each do |name, value|
+        name = name.to_s.delete_suffix(suffix) if name.to_s.end_with?(suffix)
+
+        puts "Created a(n) #{::Google::Ads::GoogleAds::Utils.camelize(name)} " \
+               "with #{value.to_s.strip}."
+      end
+  end
+end
+
+def deep_copy(o)
+  Marshal.load(Marshal.dump(o))
+end
+
+if __FILE__ == $0
+  options = {}
+  # The following parameter(s) should be provided to run the example. You can
+  # either specify these by changing the INSERT_XXX_ID_HERE values below, or on
+  # the command line.
+  #
+  # Parameters passed on the command line will override any parameters set in
+  # code.
+  #
+  # Running the example with -h will print the command line usage.
+  options[:customer_id] = 'INSERT_CUSTOMER_ID_HERE'
```

Comment:
Leave these placeholders out so command line input is the only way, and also remove them if you're modifying an example that still has them in the future. We decided to remove this going forward but not to spend the time explicitly removing it if we're not touching the example for some other reason.

Reply:
Makes sense. I copied that section. I will update others as/if go through them.